app/src/ui/changes/commit-message.tsx

```diff
@@ -246,19 +249,39 @@ export class CommitMessage extends React.Component<
   }
 
   private async createCommit() {
-    const { summary, description } = this.state
-
     if (!this.canCommit()) {
       return
     }
 
     const trailers = this.getCoAuthorTrailers()
-
-    const commitCreated = await this.props.onCreateCommit(
-      // allow single file commit without summary
+    const { summary, description } = this.state
+    // allow single file commit without summary
+    const commitSummary =
       this.props.singleFileCommit && !this.state.summary
         ? this.props.placeholder
-        : summary,
+        : summary
+
+    const lfsSupported = await this.checkForLFS()
```

Comment:
As part of rolling out new features like this, we want to enable the ability to feature-flag them, so we can control when we enable them for new users. I'm going to push a commit to this branch in a bit to let us conditionally run this check, but let me know if you have any other questions about this area:

```diff
diff --git a/app/src/lib/feature-flag.ts b/app/src/lib/feature-flag.ts
index 442d75d8d..b5ca4ff03 100644
--- a/app/src/lib/feature-flag.ts
+++ b/app/src/lib/feature-flag.ts
@@ -61,3 +61,8 @@ export function enableStatusWithoutOptionalLocks(): boolean {
 export function enableRecurseSubmodulesFlag(): boolean {
   return enableBetaFeatures()
 }
+
+/** Should the app check and warn the user about committing large files? */
+export function enableFileSizeWarningCheck(): boolean {
+  return enableDevelopmentFeatures()
+}
diff --git a/app/src/ui/changes/commit-message.tsx b/app/src/ui/changes/commit-message.tsx
index cbb71c3f0..2e1ab2b75 100644
--- a/app/src/ui/changes/commit-message.tsx
+++ b/app/src/ui/changes/commit-message.tsx
@@ -25,6 +25,7 @@ import { ITrailer } from '../../lib/git/interpret-trailers'
 import { IAuthor } from '../../models/author'
 import { IMenuItem } from '../../lib/menu-item'
 import { isUsingLFS } from '../../lib/git/lfs'
+import { enableFileSizeWarningCheck } from '../../lib/feature-flag'
 
 const addAuthorIcon = new OcticonSymbol(
   12,
@@ -261,22 +262,24 @@ export class CommitMessage extends React.Component<
         ? this.props.placeholder
         : summary
 
-    const lfsSupported = await this.checkForLFS()
-    if (lfsSupported === false) {
-      const overSizedFiles = await this.checkForLargeFiles()
-
-      if (overSizedFiles.length !== 0) {
-        this.props.dispatcher.showPopup({
-          type: PopupType.OversizedFiles,
-          dispatcher: this.props.dispatcher,
-          fileList: overSizedFiles,
-          commitSummary: commitSummary,
-          commitDescription: description,
-          repository: this.props.repository,
-          trailers: trailers,
-        })
-
-        return
+    if (enableFileSizeWarningCheck()) {
+      const lfsSupported = await this.checkForLFS()
+      if (lfsSupported === false) {
+        const overSizedFiles = await this.checkForLargeFiles()
+
+        if (overSizedFiles.length !== 0) {
+          this.props.dispatcher.showPopup({
+            type: PopupType.OversizedFiles,
+            dispatcher: this.props.dispatcher,
+            fileList: overSizedFiles,
+            commitSummary: commitSummary,
+            commitDescription: description,
+            repository: this.props.repository,
+            trailers: trailers,
+          })
+
+          return
+        }
       }
     }
```